### ISSUE TITLE
[TODO] Add a function called `get_all_headlines_as_df()`

### ISSUE THREAD

**Issue by jonjoncardoso** [jonjoncardoso/lse-ds105-week09-experiment#2]
**Deadline:** 

## Context

I want `get_all_headlines_as_df(wiki_url)` to return a pandas Data Frame whatever `wiki_url` I pass.

**Comment by uttarayand** [comment]
``` python
def get_all_headlines_as_df(link):
    response = requests.get(link)
    soup = BeautifulSoup(response.text, 'html.parser')
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'})
    return get_headlines_as_df(headlines_soup)
```

**Comment by jonjoncardoso** [comment]
> def get_all_headlines_as_df(link): response = requests.get(link) soup = BeautifulSoup(response.text, 'html.parser') headlines_soup = soup.find_all(attrs={'class': 'mw-headline'}) return get_headlines_as_df(headlines_soup)

can you add syntax highlighting?

**Comment by ajlaw138** [comment]
```python

import requests
import pandas as pd
from bs4 import BeautifulSoup

def get_headlines_as_df(wiki_url):

    response = requests.get(wiki_url)
    soup = BeautifulSoup(response.text, 'html.parser')
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'})
    df = get_headlines_as_df(headlines_soup)

  return df

```

**Comment by OjasSB** [comment]
```python
import requests
import pandas as pd
from bs4 import BeautifulSoup

def get_headlines_as_df(wiki_url):
    soup=BeautifulSoup(requests.get(wiki_url).text)
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'})
    return pd.DataFrame({'headline id': [headline.get('id') for headline in headlines_soup],
                        'parent tag': [headline.parent.name for headline in headlines_soup], 
                        'parent tag text': [headline.parent.text for headline in headlines_soup]})
```

**Comment by tz1211** [comment]
```python 
def get_all_headlines_as_df(wiki_url): 
    response = requests.get(wiki_url)
    soup = BeautifulSoup(response.text, 'html.parser') 
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'}) 
    df = get_headlines_as_df(headlines_soup) 

    return df 
```

**Comment by Peanut-butter116** [comment]
```python
def get_all_headlines_as_df(wiki_url):
    response = requests.get(wiki_url)
    soup = BeautifulSoup(response.text, 'html.parser')
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'})
    return get_headlines_as_df(headlines_soup)
```

**Comment by sashaciocirlan** [comment]
def get_all_headlines_as_df(wiki_url):
  
    response = requests.get(wiki_url)
    soup = BeautifulSoup(response.text, 'html.parser')
    headlines_soup = soup.find_all(attrs={'class': 'mw-headline'})
    
    df_headlines = pd.DataFrame({'headline_id': [headline.get('id') for headline in headlines_soup],
                                 'parent_tag': [headline.parent.name for headline in headlines_soup], 
                                 'parent_tag_text': [headline.parent.text for headline in headlines_soup]})
    return df_headlines

**Comment by jonjoncardoso** [comment]
Closing as this issue has been solved in #3
